src/lib.rs

```diff
@@ -304,15 +371,15 @@ mod getifaddrs_windows {
     // Refer: https://github.com/rust-lang/rfcs/issues/1020
     /// Return a vector of IP details for all the valid interfaces on this host
     #[allow(unsafe_code, trivial_numeric_casts)]
-    pub fn getifaddrs() -> Vec<IfAddr> {
-        let mut ret = Vec::<IfAddr>::new();
+    pub fn get_if_addrs() -> io::Result<Vec<Interface>> {
+        let mut ret = Vec::<Interface>::new();
         let mut ifaddrs: *const IpAdapterAddresses;
         let mut buffersize: c_ulong = 15000;
         loop {
             unsafe {
                 ifaddrs = libc::malloc(buffersize as size_t) as *mut IpAdapterAddresses;
                 if ifaddrs.is_null() {
-                    panic!("Failed to allocate buffer in getifaddrs()");
+                    panic!("Failed to allocate buffer in get_if_addrs()");
```

Comment:
I'm on the fence about this one. As much as I dislike `panic!` it's standard behaviour for Rust code to panic on out-of-memory.